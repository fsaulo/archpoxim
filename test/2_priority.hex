0xDC00001E
0xFC000000
0xFC000000
0xFC000000
0xDC000003
0xDC000004
0xDC000009
0xDC00000C
0xE400000F
0x80000000
0x00C00001
0x1006311E
0x48C60003
0x74C10000
0xE4000009
0x80000000
0x00C0007B
0x6CC50000
0xE4000005
0x80000000
0x00C00001
0x6CC50000
0xE4000001
0x80000000
0x2800005F
0x00200005
0x4C210001
0x5C010000
0xAFFFFFFD
0x2C0007C1
0x7C000000
0x03C07FFC
0x87E10001
0x6820002A
0x6840002B
0x6860002C
0x6880002D
0x68A0002E
0x00C00009
0x6CC50000
0x00000000
0xFC000008
0x20202020
0x20202220
0x20202221
0x20202222
0x8080888F
