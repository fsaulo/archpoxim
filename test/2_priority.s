// Segmento de codigo
.text
    // Tabela de vetor de interrupcao
    init:
        bun main
        int 0
        int 0
        int 0
        bun hw1
        bun hw2
        bun hw3
        bun hw4
        .align 5
    // Hardware 1
    hw1:
        // Procedimento de atraso
        call atraso
        // Retorno da ISR
        reti
    // Hardware 2
    hw2:
        // Hardware 1
        mov r6, 1
        sll r6, r6, 31
        addi r6, r6, 3
        s32 [r1], r6
        // Procedimento de atraso
        call atraso
        // Retorno da ISR
        reti
    // Hardware 3
    hw3:
        // Hardware 2
        mov r6, 123
        s8 [r5], r6
        // Procedimento de atraso
        call atraso
        // Retorno da ISR
        reti
    // Hardware 4
    hw4:
        // Hardware 3
        mov r6, 1
        s8 [r5], r6
        // Procedimento de atraso
        call atraso
        // Retorno da ISR
        reti
    // Procedimento de atraso
    atraso:
        // Empilhando registradores
        push r1, sr
        // Atribuindo R1 = 5
        mov r1, 5
        // Laco de espera
        subi r1, r1, 1
        cmpi r1, 0
        bat -3
        // Desempilhando registradores
        pop sr, r1
        // Retorno do procedimento
        ret
    // Funcao principal
    main:
        // SP = 32 KiB
        mov sp, 0x7FFC
        // Habilitando interrupcao (IE = 1)
        sbr sr[1]
        // Carregando enderecos de memoria
        l32 r1, [WATCHDOG]
        l32 r2, [X]
        l32 r3, [Y]
        l32 r4, [Z]
        l32 r5, [CONTROL]
        // Hardware 4
        mov r6, 9
        s8 [r5], r6
        nop
        // Finalizacao de execucao
        int 8
// Segmento de dados
.data
    // Endereco do watchdog (alinhamento de 32 bits)
    WATCHDOG:
        .4byte 0x20202020
    // Registrador X (alinhamento de 32 bits)
    X:
        .4byte 0x20202220
    // Registrador Y (alinhamento de 32 bits)
    Y:
        .4byte 0x20202221
    // Registrador Z (alinhamento de 32 bits)
    Z:
        .4byte 0x20202222
    // Registrador de controle (alinhamento de 8 bits)
    CONTROL:
        .4byte 0x8080888F
