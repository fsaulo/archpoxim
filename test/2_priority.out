[START OF SIMULATION]
0x00000000:	bun 30                   	PC=0x0000007C
0x0000007C:	mov sp,32764             	SP=0x00007FFC
0x00000080:	sbr sr[1]                	SR=0x00000002
0x00000084:	l32 r1,[r0+42]           	R1=MEM[0x000000A8]=0x20202020
0x00000088:	l32 r2,[r0+43]           	R2=MEM[0x000000AC]=0x20202220
0x0000008C:	l32 r3,[r0+44]           	R3=MEM[0x000000B0]=0x20202221
0x00000090:	l32 r4,[r0+45]           	R4=MEM[0x000000B4]=0x20202222
0x00000094:	l32 r5,[r0+46]           	R5=MEM[0x000000B8]=0x8080888F
0x00000098:	mov r6,9                 	R6=0x00000009
0x0000009C:	s8 [r5+0],r6             	MEM[0x8080888F]=R6=0x09
0x000000A0:	mov r0,0                 	R0=0x00000000
[HARDWARE INTERRUPTION 4]
0x0000001C:	bun 12                   	PC=0x00000050
0x00000050:	mov r6,1                 	R6=0x00000001
0x00000054:	s8 [r5+0],r6             	MEM[0x8080888F]=R6=0x01
0x00000058:	call 1                   	PC=0x00000060,MEM[0x00007FF0]=0x0000005C
[HARDWARE INTERRUPTION 3]
0x00000018:	bun 9                    	PC=0x00000040
0x00000040:	mov r6,123               	R6=0x0000007B
0x00000044:	s8 [r5+0],r6             	MEM[0x8080888F]=R6=0x7B
0x00000048:	call 5                   	PC=0x00000060,MEM[0x00007FE0]=0x0000004C
[HARDWARE INTERRUPTION 2]
0x00000014:	bun 4                    	PC=0x00000028
0x00000028:	mov r6,1                 	R6=0x00000001
0x0000002C:	sll r0,r6,r6,30          	R0:R6=R0:R6<<31=0x0000000080000000,SR=0x00000002
0x00000030:	addi r6,r6,3             	R6=R6+0x00000003=0x80000003,SR=0x00000012
0x00000034:	s32 [r1+0],r6            	MEM[0x80808080]=R6=0x80000003
0x00000038:	call 9                   	PC=0x00000060,MEM[0x00007FD0]=0x0000003C
0x00000060:	push r1,sr               	MEM[0x00007FCC]{0x20202020,0x00000012}={R1,SR}
0x00000064:	mov r1,5                 	R1=0x00000005
[HARDWARE INTERRUPTION 1]
0x00000010:	bun 3                    	PC=0x00000020
0x00000020:	call 15                  	PC=0x00000060,MEM[0x00007FB8]=0x00000024
0x00000060:	push r1,sr               	MEM[0x00007FB4]{0x00000005,0x00000012}={R1,SR}
0x00000064:	mov r1,5                 	R1=0x00000005
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000004,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000003,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000002,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000001,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000000,SR=0x00000042
0x0000006C:	cmpi r1,0                	SR=0x00000042
0x00000070:	bat -3                   	PC=0x00000074
0x00000074:	pop sr,r1                	{SR,R1}=MEM[0x00007FAC]{0x00000012,0x00000005}
0x00000078:	ret                      	PC=MEM[0x00007FB8]=0x00000024
0x00000024:	reti                     	IPC=MEM[0x00007FBC]=0x00000048,CR=MEM[0x00007FC0]=0x01EEE754,PC=MEM[0x00007FC4]=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000004,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000003,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000002,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000001,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000000,SR=0x00000042
0x0000006C:	cmpi r1,0                	SR=0x00000042
0x00000070:	bat -3                   	PC=0x00000074
0x00000074:	pop sr,r1                	{SR,R1}=MEM[0x00007FC4]{0x00000012,0x20202020}
0x00000078:	ret                      	PC=MEM[0x00007FD0]=0x0000003C
0x0000003C:	reti                     	IPC=MEM[0x00007FD4]=0x00000058,CR=MEM[0x00007FD8]=0x01EEE754,PC=MEM[0x00007FDC]=0x00000060
0x00000060:	push r1,sr               	MEM[0x00007FDC]{0x20202020,0x00000012}={R1,SR}
0x00000064:	mov r1,5                 	R1=0x00000005
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000004,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000003,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000002,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000001,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000000,SR=0x00000042
0x0000006C:	cmpi r1,0                	SR=0x00000042
0x00000070:	bat -3                   	PC=0x00000074
0x00000074:	pop sr,r1                	{SR,R1}=MEM[0x00007FD4]{0x00000012,0x20202020}
0x00000078:	ret                      	PC=MEM[0x00007FE0]=0x0000004C
0x0000004C:	reti                     	IPC=MEM[0x00007FE4]=0x000000A0,CR=MEM[0x00007FE8]=0x01EEE754,PC=MEM[0x00007FEC]=0x00000060
0x00000060:	push r1,sr               	MEM[0x00007FEC]{0x20202020,0x00000012}={R1,SR}
0x00000064:	mov r1,5                 	R1=0x00000005
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000004,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000003,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000002,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000001,SR=0x00000002
0x0000006C:	cmpi r1,0                	SR=0x00000002
0x00000070:	bat -3                   	PC=0x00000068
0x00000068:	subi r1,r1,1             	R1=R1-0x00000001=0x00000000,SR=0x00000042
0x0000006C:	cmpi r1,0                	SR=0x00000042
0x00000070:	bat -3                   	PC=0x00000074
0x00000074:	pop sr,r1                	{SR,R1}=MEM[0x00007FE4]{0x00000012,0x20202020}
0x00000078:	ret                      	PC=MEM[0x00007FF0]=0x0000005C
0x0000005C:	reti                     	IPC=MEM[0x00007FF4]=0x00000000,CR=MEM[0x00007FF8]=0x00000000,PC=MEM[0x00007FFC]=0x000000A4
0x000000A4:	int 8                    	CR=0x00000008,PC=0x0000000C
[SOFTWARE INTERRUPTION]
0x0000000C:	int 0                    	CR=0x00000000,PC=0x00000000
[END OF SIMULATION]
